ci/test_cpp.sh

```diff
@@ -9,7 +9,23 @@ rapids-logger "Generate C++ testing dependencies"
 rapids-dependency-file-generator \
   --output conda \
   --file_key test_cpp \
-  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch)" | tee env.yaml
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch)" | tee test.yaml
+
+rapids-logger "Downloading artifacts from previous jobs"
+CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
+
+cat <<EOF | tee cuml.yaml
+name: cuml
+channels:
+  - ${CPP_CHANNEL}
+dependencies:
+  - libcuml
+  - libcuml-tests
+EOF
+
+# TODO: install conda-merge in the CI images
```

Comment:
I created a PR to the CI images for this: https://github.com/rapidsai/ci-imgs/pull/112 but that would be unnecessary if we adopt this alternative approach: https://github.com/rapidsai/cuml/pull/5781#issuecomment-1958498274